packages/trilinoscouplings/examples/scaling/TrilinosCouplings_TpetraIntrepidHybridPoisson2DExample.cpp

```diff
@@ -1118,39 +1118,37 @@ makeMatrixAndRightHandSide (Teuchos::RCP<sparse_matrix_type>& A,
     // Import from the global column map to the local column map.
     myColsToZeroT->doImport (*globColsToZeroT, *bdyExporter, Tpetra::INSERT);
 
-    Array<ST> values;
-    Array<int> indices;
     ArrayRCP<const int> myColsToZeroArrayRCP = myColsToZeroT->getData(0);
     size_t NumEntries = 0;
 
     // Zero the columns corresponding to Dirichlet BCs.
     for (LO i = 0; i < as<int> (gl_StiffMatrix->getNodeNumRows ()); ++i) {
       NumEntries = gl_StiffMatrix->getNumEntriesInLocalRow (i);
-      values.resize (NumEntries);
-      indices.resize (NumEntries);
-      gl_StiffMatrix->getLocalRowCopy (i, indices (), values (), NumEntries);
+      typename sparse_matrix_type::nonconst_local_inds_host_view_type indices("indicies", NumEntries);
+      typename sparse_matrix_type::nonconst_values_host_view_type values("values", NumEntries);
+      gl_StiffMatrix->getLocalRowCopy (i, indices, values, NumEntries);
       for (int j = 0; j < as<int> (NumEntries); ++j) {
-        if (myColsToZeroArrayRCP[indices[j]] == 1)
-          values[j] = STS::zero ();
+        if (myColsToZeroArrayRCP[indices(j)] == 1)
+          values(j) = STS::zero ();
       }
-      gl_StiffMatrix->replaceLocalValues (i, indices (), values ());
+      gl_StiffMatrix->replaceLocalValues (i, indices, values);
     } // for each (local) row of the global stiffness matrix
 
     // Zero the rows and add ones to diagonal.
     for (int i = 0; i < numBCNodes; ++i) {
       NumEntries = gl_StiffMatrix->getNumEntriesInLocalRow (BCNodes[i]);
-      indices.resize (NumEntries);
-      values.resize (NumEntries);
-      gl_StiffMatrix->getLocalRowCopy (BCNodes[i], indices (), values (), NumEntries);
+      typename sparse_matrix_type::nonconst_local_inds_host_view_type indices("indicies", NumEntries);
+      typename sparse_matrix_type::nonconst_values_host_view_type values("values", NumEntries);
```

Comment:
A quick question. The old code used resize, so that it would only allocate if the current capacity was not big enough. This change now always allocates new views and could result in a slow down on this algorithm. Not sure if that is important, in this context. Notice this pattern is repeated a few times below.

Reply:
Btw, "indicies" is misspelled.

Reply:
I agree with @rppawlo, this is could be a bottleneck.

Reply:
do views have a resize?  I'll look, i was thinking it isn't a huge deal with this code, but should be fixed.

Reply:
Went to Kokkos::resize(view, newsize) calls everywhere.